conftest.py

```diff
@@ -1,2 +1,201 @@
+    pxe_template = entities.ConfigTemplate().search(
+        query={
+            u'search': u'name="{0}"'.format(DEFAULT_PXE_TEMPLATE)
+        }
+    )
+    pxe_template = pxe_template[0].read()
+    pxe_template.organization.append(module_org)
+    pxe_template.location.append(module_location)
+    pxe_template.update(['organization', 'location'])
+    pxe_template = entities.ConfigTemplate(id=pxe_template.id).read()
+    return pxe_template
+
+
+@pytest.fixture(scope='module')
+def module_architecture():
+    arch = entities.Architecture().search(
+        query={
+            u'search': u'name="{0}"'.format(DEFAULT_ARCHITECTURE)
+        }
+    )[0].read()
+    return arch
+
+
+@pytest.fixture(scope='module')
+def module_os(
+        module_architecture, module_partiontable, module_configtemaplate,
+        module_provisioingtemplate, os=None):
+    if os is None:
+        os = entities.OperatingSystem().search(query={
+            u'search': u'name="RedHat" AND (major="{0}" OR major="{1}")'.format(
+                RHEL_6_MAJOR_VERSION, RHEL_7_MAJOR_VERSION)
+        })[0].read()
+    else:
+        os = entities.OperatingSystem().search(query={
+            u'search': u'family="Redhat" AND major="{0}" AND minor="{1}")'.format(
+                os.split(' ')[1].split('.')[0], os.split(' ')[1].split('.')[1])
+        })[0].read()
+    os.architecture.append(module_architecture)
+    os.ptable.append(module_partiontable)
+    os.config_template.append(module_configtemaplate)
+    os.provisioning_template.append(module_provisioingtemplate)
+    os.update([
+        'architecture',
+        'ptable',
+        'config_template',
+        'provisioning_template'
+    ])
+    os = entities.OperatingSystem(id=os.id).read()
+    return os
+
+
+@pytest.fixture(scope='module')
+def module_puppet_environment(module_org, module_location):
+    environments = entities.Environment().search(
+        query=dict(search='organization_id={0}'.format(module_org.id)))
+    if len(environments) > 0:
+        environment = environments[0].read()
+        environment.location.append(module_location)
+        environment = environment.update(['location'])
+    else:
+        environment = entities.Environment(
+            organization=[module_org],
+            location=[module_location]
+        ).create()
+    puppetenv = entities.Environment(id=environment.id).read()
+    return puppetenv
+
+
+# Google Cloud Engine Entities
+@pytest.fixture(scope='session')
+def googleclient():
+    gceclient = GoogleCloudSystem(
+        project=settings.gce.project_id,
+        zone=settings.gce.zone,
+        file_path=download_gce_cert(),
+        file_type='json')
+    yield gceclient
+    gceclient.disconnect()
+
+
+@pytest.fixture(scope='session')
+def gce_latest_rhel_uuid(googleclient):
+    max_rhel7_template = max(
+        [img.name for img in googleclient.list_templates(
+            True) if str(img.name).startswith('rhel-7')])
```

Comment:
List comprehension doesn't really serve readability of this. I would break it into two lines:
```python
template_names = [img.name for img in googleclient.list_templates(True)]
latest_rhel7_template = max(name for name in template_names if name.startswith('rhel-7'))
```

Reply:
Again, I dont waste variable if its not reusable, but updating for better understanding.